sequencing/ATACSeq/Rscripts/collateSampleChecks.Rmd

```diff
@@ -262,34 +262,37 @@ We will define candidates for potential swaps as those with > 0.9 contamination
 # QC Summary: Sample Filtering
 
 ```{r samplePASS}
-# generate a txt file of passed samples
+
 QCPASS<-cbind(sexPredict$concordantLabel, genoScores$genoCheck)
 colnames(QCPASS)<-c("sexCheck", "genoCheck")
 pheno <- pheno[match(sexPredict$sampleID, pheno$sampleID),]
 passStage2<- cbind(pheno, QCPASS)
 passStage2$genoCheck[is.na(passStage2$genoCheck)& passStage2$sexCheck == TRUE]<- TRUE
-write.csv(passStage2, file = file.path(metaDir, "/passStage2Status.csv"))
 
-# need to keep samples without sex or genotype data to confirm.
-keep<-rowSums(QCPASS) == rowSums(!is.na(QCPASS))
+passStage2<-cbind(passStage2, "QCPASS" = rowSums(QCPASS) == ncol(QCPASS))
+passStage2 <- as.data.frame(passStage2)
+passStage2$QCPASS[passStage2$genoCheck == TRUE & passStage2$sexCheck == TRUE]<- TRUE
+passStage2$QCPASS[is.na(passStage2$genoCheck)& passStage2$sexCheck == TRUE]<- TRUE
+passStage2$QCPASS[is.na(passStage2$genoCheck)& passStage2$sexCheck == FALSE]<- FALSE
+passStage2$QCPASS[is.na(passStage2$sexCheck)& passStage2$genoCheck == FALSE]<- FALSE
+passStage2$QCPASS[is.na(passStage2$sexCheck)& passStage2$genoCheck == TRUE]<- TRUE
+passStage2$QCPASS[is.na(passStage2$sexCheck)& is.na(passStage2$genoCheck)]<- FALSE

```

Comment:
This could be simplified down to:

```R
passStage2$QCPASS <- (
  passStage2$genoCheck == TRUE |
  passStage2$sexCheck == TRUE) | 
  (is.na(passStage2$genoCheck) & is.na(passStage2$sexCheck))
```

Which might also be a little clearer.

Reply:
Yeah, I though it was a bit messy

Reply:
You could rewrite it however, you wish. The point is, what you care about is:

- Is genoCheck TRUE
- Is sexCheck TRUE
- Or are both not there

Reply:
Please check this as I might be wrong. I'm pretty sure that's what this is trying to do. I've already had to change the code I put twice.

Reply:
There are too many "if else" that I didn't know how to collapse them in  less lines of codes